salt/states/zfs.py

```diff
@@ -706,4 +706,217 @@ def promoted(name):
 
     return ret
 
+
+def scheduled_snapshot(name, prefix, recursive=True, schedule=None):
+    '''
+    create snapshots based on a schedule
+
+    name : string
+        name of filesystem or volume
+    prefix : string
+        prefix for the snapshots
+        e.g. 'test' will result in snapshots being named 'test-YYYYMMDD_HHMM'
+    recursive : boolean
+        create snapshots for all children also
+    schedule : dict
+        dict holding the schedule, the following keys are available (minute, hour,
+        day, month, and year) by default all are set to 0 the value indicated the
+        number of snapshots of that type to keep around.
+
+    ..warning::
```

Comment:
Ah, OK. The description in the docstring confused me but what you're saying makes perfect sense. Thanks for the clarification.